src/lib.rs

```diff
@@ -159,14 +159,17 @@ where
 
         let v = node.value.load(Ordering::SeqCst, guard);
         assert!(!v.is_null());
-        Some(v)
+        // safety: the lifetime of the reference is bound to the guard
+        // supplied which means that the memory will not be modified
+        // until at least after the guard goes out of scope
+        unsafe { v.as_ref() }
     }
 
     pub fn get_and<R, F: FnOnce(&V) -> R>(&self, key: &K, then: F) -> Option<R> {
```

Comment:
Arguably, `get_and` doesn't need to exist now

Reply:
Feel free to remove it!

Reply:
I think given your comment with regards to an API which doesn't require a guard to be supplied, I think for now I'll leave it. I have some ideas of how to provide a more flexible API in this regard so I might raise that as a separate PR